### PR TITLE
Add the TLSv1_3 setting to disable TLSv1_3

### PR DESCRIPTION
**Pull request by shinrich**
Added the TLSv1_3 config setting like the other per-protocol settings to allow for disabling TLSv1.3 (enabled by default).

Updated the docs and updates one of the tls tests to exercise the new setting.

### CODE REVIEW

**Comment by zwoop** [comment]
I think this looks good, but we should have @maskit review as well.

**Comment by maskit** [comment]
I'm fine with adding this setting, but I'm curious how it works in the future and we should probably clarify what this setting is for; Whether this applies to QUIC connections.

Because the first version of QUIC would require TLS 1.3, it would not work if TLS 1.3 is disabled entirely on ATS, which is fine with me if we do it intentionally. I'll add some note about it on the documentation.

Another interpretation of this setting is that this is only for ordinary SSL/TLS connections over TCP, which means it doesn't work for QUIC connections. If we go this way, we can disable TLS 1.3 on the TCP connections but we can still use QUIC. I'm not sure if we want such a configuration but it doesn't sound crazy idea to me. If newer versions of QUIC supported multiple versions of TLS, we would need one more setting, "proxy.config.*quic*.TLSv1_3".

**Comment by shinrich** [comment]
I hadn't considered the QUIC application, but there are quite a few TLS settings that may or may not be applicable for QUIC (e.g. cipher selections, certificate verifications).  We probably want to take a separate pass and determine how to normalize those.

File: doc/admin-guide/files/records.config.en.rst [hunk 1]
```diff
@@ -3245,6 +3245,10 @@ SSL Termination
 
    Enables (``1``) or disables (``0``) TLS v1.2.  If not specified, enabled by default.  [Requires OpenSSL v1.0.1 and higher]
 
+.. ts:cv:: CONFIG proxy.config.ssl.TLSv1_3 INT 1
+
+   Enables (``1``) or disables (``0``) TLS v1.3.  If not specified, enabled by default.  [Requires OpenSSL v1.1.1 and higher]
```
Comment (SolidWallOfCode) [inline doc/admin-guide/files/records.config.en.rst]: WRT to @maskit this should be updated to note that QUIC depends on TLS 1.3 or better.

**Comment by maskit** [comment]
> Even if TLS v1.3 is off, if I explicitly enable QUIC, I get TLS v1.3 but only for the UDP ports. I think anyone else is asking for trouble and confusion.

While I understand the needs, I think it should be a separated setting for each transport layer. I'd say this one is for TLS engine.

**Comment by mlibbey** [comment]
Since this commit, autest is failing in various tests for most new PRs -- but, each on the "SNI message appears only once" part (which appears in multiple tls_hooks/tls_hooks[247].py).

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**